src/views/ApartmentApp/components/StudentApplication/components/ApplicantList/components/ApplicantListItem/index.js

```diff
@@ -0,0 +1,154 @@
+import user from '../../../../../../../../services/user';
+
+/**
+ * @typedef { import('../../../../services/user').StudentProfileInfo } StudentProfileInfo
+ */
+
+// Based off src/views/PeopleSearch/components/PeopleSearchResult
+// but using props.profile of type StudentProfileInfo
+// rather than using Person of type PeopleSearchResult
+const ApplicantListItem = ({
+  disabled,
+  profile,
+  isApplicationEditor,
+  onChangeEditor,
+  onApplicantRemove,
+}) => {
+  const [avatar, setAvatar] = useState(null);
+  const [hasNickName, setHasNickname] = useState(false);
+  const [personClass, setPersonClass] = useState(profile.Class);
+
+  useEffect(() => {
+    loadAvatar(profile);
+    createNickname(profile);
+    if (String(profile.PersonType).includes('stu') && profile.Class !== undefined) {
+      setPersonClass(profile.Class);
+    } else {
+      // Technically, this case should never happen because the list does not allow the user to add a non-student to the applicant list
+      setPersonClass('');
+    }
+  }, [profile]);
+
+  /**
+   * Creates the Avatar image of the given user
+   * @param {StudentProfileInfo} profile The StudentProfileInfo object for the student represented by this list item
+   */
+  const loadAvatar = async (profile) => {
+    try {
+      const { def: defaultImage, pref: preferredImage } = await user.getImage(profile.AD_Username);
+      setAvatar(preferredImage || defaultImage);
+    } catch {
+      setAvatar(
+        <svg width="50" height="50" viewBox="0 0 50 50">
+          <rect width="50" height="50" rx="10" ry="10" fill="#CCC" />
+        </svg>,
+      );
+    }
+  };
+
+  /**
+   * Saves the nickname of the given user if available
+   * @param {StudentProfileInfo} profile The StudentProfileInfo object for the student represented by this list item
+   */
+  const createNickname = (profile) => {
+    let Name = String(profile.fullName);
+    let FirstName = Name.split(' ')[0];
+    setHasNickname(FirstName !== profile.NickName && profile.NickName !== '');
+  };
+
+  /**
+   * Callback for changing the application editor
+   * @param {StudentProfileInfo} profile The StudentProfileInfo object for the person who is to be made the application editor
+   */
+  const handleChangeEditor = () => {
+    // Make sure the chosen profile was not null
+    if (profile) {
+      // Send the selected profile to the parent component
+      onChangeEditor(profile);
+    }
+  };
+
+  /**
+   * Callback for applicant list remove button
+   * @param {StudentProfileInfo} profileToRemove The StudentProfileInfo object for the person who is to be removed from the list of applicants
+   */
+  const handleRemove = () => {
+    // Make sure the chosen profile was not null
+    if (profile) {
+      // Send the selected profile to the parent component
+      onApplicantRemove(profile);
+    }
+  };
+
+  const displayName = hasNickName
+    ? profile.fullName + ' (' + profile.NickName + ')'
+    : profile.fullName;
```

Comment:
Since `fullName` is added by `user.js` in the frontend, you can't depend on having it. If any of the listed applicants were loaded from the backend rather than being added by the frontend, they will not have this property. Your two options are two add the property yourself in the backend, or to sinply use First and Last name normally. I would recommend the latter.
```suggestion
    ? `${profile.FirstName} ${profile.LastName} (${profile.NickName})`
    : `${profile.FirstName} ${profile.LastName}`;
```

Reply:
I agree, feel free to commit that suggestion since I do not have write access to this branch